aya/src/programs/mod.rs

```diff
@@ -540,6 +552,47 @@ impl<'a, P: ProgramFd> ProgramFd for &'a P {
     }
 }
 
+macro_rules! impl_program_unload {
+    ($($struct_name:ident),+ $(,)?) => {
+        $(
+            impl $struct_name {
+                /// Unloads the program from the kernel.
+                ///
+                /// If `detach` is true, links will be detached before unloading the program.
+                /// Note that OwnedLinks you obtained using [KProbe::forget_link] will not be detached.
```

Comment:
`s/KProbe/$struct_name/g`

Reply:
ok

Reply:
I don't think you can expand struct_name in the comment.

I'd go for "Note that owned links obtained using `forget_link()` will not be detached."